common/components/Header/components/NetworkSelector/NetworkOption.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { translateRaw } from 'translations';
+import classnames from 'classnames';
+import { isAutoNode, isAutoNodeConfig } from 'libs/nodes';
+import { NodeConfig } from 'types/node';
+import { NetworkConfig } from 'types/network';
+import NodeOption from './NodeOption';
+import './NetworkOption.scss';
+
+interface Props {
+  nodes: NodeConfig[];
+  network: NetworkConfig;
+  nodeSelection: string;
+  isNetworkSelected: boolean;
+  isExpanded: boolean;
+  selectNode(node: NodeConfig): void;
```

Comment:
Another cool thing, but not related: you can get the method types of a class by using accessor syntax so you dont have to re-write these. E.g ```ts
import {NetworkSelector /* Assuming you've exported the non-connected version of the class */}
interface Props {
 selectNode: NetworkSelector["selectNode"]
}
```

Reply:
Definitely makes refactoring easier, but it does introduce more cyclic dependencies, which eventually I'd like to get rid of.

Reply:
Yeah, only way to avoid that is by having the class implement an interface, but at that point its a little verbose